src/index.tsx

```diff
@@ -20,122 +25,136 @@ export default class SignatureCanvas extends Component {
     clearOnResize: PropTypes.bool
   }
 
-  static defaultProps = {
+  static defaultProps: Pick<ReactSignatureCanvasProps, 'clearOnResize'> = {
     clearOnResize: true
   }
 
-  _sigPad = null
+  // this is some hack-ish init and casting to avoid `| null` everywhere :/
+  /* eslint-disable @typescript-eslint/consistent-type-assertions */
+  _sigPad: SignaturePad = {} as SignaturePad
+  _canvas: HTMLCanvasElement = {} as HTMLCanvasElement
+  /* eslint-enable @typescript-eslint/consistent-type-assertions */
+
+  _setRef = (ref: HTMLCanvasElement | null): void => {
+    if (ref !== null) {
+      this._canvas = ref
+    }
+  }
 
-  _excludeOurProps = () => {
+  _excludeOurProps = (): SignaturePad.SignaturePadOptions => {
     const { canvasProps, clearOnResize, ...sigPadProps } = this.props
     return sigPadProps
   }
 
-  componentDidMount () {
+  componentDidMount: Component['componentDidMount'] = () => {
     this._sigPad = new SignaturePad(this._canvas, this._excludeOurProps())
     this._resizeCanvas()
     this.on()
   }
 
-  componentWillUnmount () {
+  componentWillUnmount: Component['componentWillUnmount'] = () => {
     this.off()
   }
 
   // propagate prop updates to SignaturePad
-  componentDidUpdate () {
+  componentDidUpdate: Component['componentDidUpdate'] = () => {
     Object.assign(this._sigPad, this._excludeOurProps())
   }
 
   // return the canvas ref for operations like toDataURL
-  getCanvas = () => {
+  getCanvas = (): HTMLCanvasElement => {
     return this._canvas
   }
 
   // return a trimmed copy of the canvas
-  getTrimmedCanvas = () => {
+  getTrimmedCanvas = (): HTMLCanvasElement => {
     // copy the canvas
     const copy = document.createElement('canvas')
     copy.width = this._canvas.width
     copy.height = this._canvas.height
-    copy.getContext('2d').drawImage(this._canvas, 0, 0)
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    copy.getContext('2d')!.drawImage(this._canvas, 0, 0)
     // then trim it
     return trimCanvas(copy)
   }
 
   // return the internal SignaturePad reference
-  getSignaturePad = () => {
+  getSignaturePad = (): SignaturePad => {
     return this._sigPad
   }
 
-  _checkClearOnResize = () => {
+  _checkClearOnResize = (): void => {
     if (!this.props.clearOnResize) {
       return
     }
     this._resizeCanvas()
   }
 
-  _resizeCanvas = () => {
-    const canvasProps = this.props.canvasProps || {}
+  // a few eslint disables of strict-boolean-expressions here because changing
+  // these lines would be breaking -- 0s, '', null etc are falsey
+  _resizeCanvas = (): void => {
+    const canvasProps = this.props.canvasProps ?? {}
     const { width, height } = canvasProps
```

Comment:
These two lines could be merged, `canvasProps` const is not used anywhere.
```suggestion
    const { width, height } = this.props.canvasProps ?? {}
```

Reply:
https://github.com/agilgur5/react-signature-canvas/pull/42#discussion_r376749731